src/servicesHost.ts

```diff
@@ -97,13 +108,15 @@ export function makeServicesHost(
     /**
      * For @types expansion, these two functions are needed.
      */
-    directoryExists: compiler.sys.directoryExists,
+    directoryExists: moduleResolutionHost.directoryExists,
 
     useCaseSensitiveFileNames: () => compiler.sys.useCaseSensitiveFileNames,
 
+    realpath: moduleResolutionHost.realpath,
```

Comment:
We didn't have `realpath` before I think; I'm just curious; what is it used for?

Reply:
Ah - we've had since we added symlink support here: https://github.com/TypeStrong/ts-loader/pull/774/files

We just didn't supply it to the `LanguageServiceHost` until now though. I'm kind of surprised we didn't get a compilation error previously 🤔

Reply:
Hm, interesting. Perhaps https://github.com/Microsoft/TypeScript/pull/12020#issuecomment-396287996 (and the whole PR) is related to this.